generator/templates/contrib/stratoscale/README.md

```diff
@@ -0,0 +1,311 @@
+* `apiKey` - token that should be able to processed.
+* `oauth2` - token and scopes that should be processed. 
+* and `basic` - user/password that should be processed.
+
+Here we defined an apiKey, that is passed through the Cookie header.
+
+The `security` section defines the default security enforcement for the application. You can select
+different securityDefinitions, as the keys, and apply "scopes" as the values. Those default definitions
+can be overriden in each route by a section with the same name:
+
+```yaml
+paths:
+  /pets:
+    post:
+      [...]
+      security:
+        - token: [admin]
+```
+
+Here we overriden the scope of token in the POST /pets URL so that only admin can use this API.
+
+Let's see how we can use this functionality.
+
+## Writing Security Handlers
+
+Once we created a security definition named "token", a function called "AuthToken" was added to the `restapi.Config`:
+
+```go
+type Config struct {
+    ...
+	// AuthToken Applies when the "Cookie" header is set
+	AuthToken func(token string) (interface{}, error)
+}
+```
+
+This function gets the content of the Cookie header, and should return an `interface{}` and `error`.
+The `interface{}` is the object that should represent the user that performed the request, it should
+be nil to return an unauthorized 401 HTTP response. If the returned `error` is not nil, an HTTP 500,
+internal server error will be returned.
+
+The returned object, will be stored in the request context under the `restapi.AuthKey` key.
+
+There is another function that we should know about, in the `restapi.Config` struct:
+
+```go
+type Config struct {
+	...
+	// Authorizer is used to authorize a request after the Auth function was called using the "Auth*" functions
+	// and the principal was stored in the context in the "AuthKey" context value.
+	Authorizer func(*http.Request) error
+}
+```
+
+This one is a custom defined function that gets the request and can return an error.
+If the returned error is not nil, and 403 HTTP error will be returned to the client - here the policy
+enforcement comes to place.
+There are two things that this function should be aware of:
+
+1. The user - it can retrieve the user information from the context: `ctx.Value(restapi.AuthKey).(MyUserType)`.
+   Usually, a server will have a function for extracting this user information and returns a concrete
+   type which could be used by all the routes.
+2. The route - it can retrieve the route using the go-swagger function: `middleware.MatchedRouteFrom(*http.Request)`.
+   So no need to parse URL and test the request method.
+   This route struct contains the route information. If for example, we want to check the scopes that were
+   defined for the current route in the swagger.yaml we can use the code below:
+   
+```go
+for _, auth := range route.Authenticators {
+    for scopeName, scopeValues := range auth.Scopes {
+        for _, scopeValue := range scopeValues {
+            ...
+        }
+    }
+}
+```
```

Comment:
This markdown would be better located in the `./examples` directory.
You are welcome to add some `example/contributed-templates/stratoscale` directory.
Would be nice to add in there code for a full generated example.
The maintenance of examples (regeneration)) is normally carried out manually from time to time using the script in `hack/regen-sample.sh`. Please add your sample doc and code and update this script so we can maintain the example up-to-date with every maintenane.